### PR TITLE
yaml-cpp*: Refactor, rename, adopt

### PR DESCRIPTION
**Pull request by OPNA2608**
###### Description of changes

- Refactored `libyamlcpp` and `libyamlcpp_0_3` abit
  - Enabled tests
    - Checked that they pass on x86_64-linux
    - Checked that they're not being built/run under cross-compilation
  - Enabled on more platforms
    - Tested with `pkgsCross.mingwW64.libyamlcpp`, `pkgsCross.mingwW64.libyamlcpp_0_3`
- Renamed the packages so the names match upstream's project name & [the name in other package repos](https://repology.org/project/yaml-cpp/versions)
  - `libyamlcpp` -> `yaml-cpp`
  - `libyamlcpp_0_3` -> `yaml-cpp_0_3`
- Adopted these packages
  - It'll be needed for some Lomiri stuff :)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by OPNA2608** [comment]
@ofborg build yaml-cpp yaml-cpp_0_3

File: pkgs/development/libraries/yaml-cpp/default.nix [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      url = "https://github.com/jbeder/yaml-cpp/commit/328d2d85e833be7cb5a0ab246cc3f5d7e16fc67a.patch";
+      hash = "sha256-1M2rxfbVOrRH9kiImcwcEolXOP8DeDW9Cbu03+mB5Yk=";
+    })
+  ];
+
+  postPatch = ''
+    # Don't install vendored gtest when running tests
+    substituteInPlace test/CMakeLists.txt \
+      --replace 'CMAKE_CURRENT_BINARY_DIR}/prefix"' 'CMAKE_CURRENT_BINARY_DIR}/prefix" EXCLUDE_FROM_ALL'
+  '';
```
Comment (tobim) [inline pkgs/development/libraries/yaml-cpp/default.nix:40]: I think `"-DINSTALL_GTEST=OFF"` in `cmakeFlags` should have the same effect.

Comment (OPNA2608) [reply pkgs/development/libraries/yaml-cpp/default.nix:40]: You're right! That's easier for our purposes, thanks.

File: pkgs/development/libraries/yaml-cpp/0.3.0.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  cmakeFlags = [
+    "-DYAML_CPP_BUILD_TOOLS=${lib.boolToString doCheck}"
+    "-DBUILD_SHARED_LIBS=${lib.boolToString (!stdenv.hostPlatform.isStatic)}"
+  ];
+
+  doCheck = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
+
+  meta = with lib; {
+    inherit (src.meta) homepage;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/yaml-cpp/0.3.0.nix:32]: Please write the url literal.

File: pkgs/development/libraries/yaml-cpp/default.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  ];
+
+  doCheck = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
+
+  passthru.updateScript = gitUpdater {
+    rev-prefix = "yaml-cpp-";
+  };
+
+  meta = with lib; {
+    inherit (src.meta) homepage;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/yaml-cpp/default.nix:55]: same

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 213490` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>fcitx-engines.rime</li>
    <li>linuxKernel.packages.hardkernel_4_14.sysdig</li>
    <li>linuxPackages_hardkernel_latest.sysdig</li>
    <li>luxcorerender</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>linuxPackages_4_14_hardened.sysdig (linuxKernel.packages.linux_4_14_hardened.sysdig)</li>
    <li>linuxPackages_4_19_hardened.sysdig (linuxKernel.packages.linux_4_19_hardened.sysdig)</li>
    <li>linuxPackages_testing_bcachefs.sysdig (linuxKernel.packages.linux_testing_bcachefs.sysdig)</li>
    <li>python311Packages.validphys2</li>
  </ul>
</details>
<details>
  <summary>65 packages built:</summary>
  <ul>
    <li>ashuffle</li>
    <li>blender</li>
    <li>blender-hip</li>
    <li>brise</li>
    <li>calamares</li>
    <li>calamares-nixos</li>
    <li>contour</li>
    <li>coreboot-configurator</li>
    <li>d-seams</li>
    <li>embree</li>
    <li>embree2</li>
    <li>facter</li>
    <li>fcitx5-rime</li>
    <li>huggle</li>
    <li>ibus-engines.rime</li>
    <li>interception-tools</li>
    <li>ip2unix</li>
    <li>jami-client</li>
    <li>jami-daemon</li>
    <li>librime</li>
    <li>linuxKernel.packages.linux_4_14.sysdig</li>
    <li>linuxKernel.packages.linux_4_19.sysdig</li>
    <li>linuxKernel.packages.linux_5_10.sysdig</li>
    <li>linuxPackages_5_10_hardened.sysdig (linuxKernel.packages.linux_5_10_hardened.sysdig)</li>
    <li>linuxPackages.sysdig (linuxKernel.packages.linux_5_15.sysdig)</li>
    <li>linuxPackages_hardened.sysdig (linuxPackages_5_15_hardened.sysdig)</li>
    <li>linuxKernel.packages.linux_5_4.sysdig</li>
    <li>linuxPackages_latest.sysdig (linuxKernel.packages.linux_6_1.sysdig)</li>
    <li>linuxPackages_6_1_hardened.sysdig (linuxKernel.packages.linux_6_1_hardened.sysdig)</li>
    <li>linuxPackages_latest-libre.sysdig (linuxKernel.packages.linux_latest_libre.sysdig)</li>
    <li>linuxPackages-libre.sysdig (linuxKernel.packages.linux_libre.sysdig)</li>
    <li>linuxPackages_lqx.sysdig (linuxKernel.packages.linux_lqx.sysdig)</li>
    <li>linuxPackages_xanmod.sysdig (linuxKernel.packages.linux_xanmod.sysdig)</li>
    <li>linuxPackages_xanmod_latest.sysdig (linuxKernel.packages.linux_xanmod_latest.sysdig ,linuxPackages_xanmod_stable.sysdig)</li>
    <li>linuxPackages_zen.sysdig (linuxKernel.packages.linux_zen.sysdig)</li>
    <li>luaformatter</li>
    <li>micromamba</li>
    <li>mir</li>
    <li>mongodb (mongodb-6_0)</li>
    <li>mongodb-4_2</li>
    <li>mongodb-4_4</li>
    <li>mongodb-5_0</li>
    <li>nnpdf (python310Packages.nnpdf)</li>
    <li>opencolorio</li>
    <li>openimageio</li>
    <li>openxcom</li>
    <li>osl</li>
    <li>osm2xmap</li>
    <li>pdns</li>
    <li>python310Packages.bpycv</li>
    <li>validphys2 (python310Packages.validphys2)</li>
    <li>python311Packages.nnpdf</li>
    <li>rime-data</li>
    <li>rstudio</li>
    <li>rstudio-server</li>
    <li>rstudioServerWrapper</li>
    <li>rstudioWrapper</li>
    <li>sysdig</li>
    <li>thinkfan</li>
    <li>timeloop</li>
    <li>torrenttools</li>
    <li>way-displays</li>
    <li>xfitter</li>
    <li>yaml-cpp</li>
    <li>yaml-cpp_0_3</li>
  </ul>
</details>

**Pull request merged**